screens/user-signup/user-signup.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react';
+import { ActivityIndicator, Button, Platform, Pressable, StyleSheet, Text, View } from 'react-native';
+import { UserRoler } from '../../components/serializes/user-roles';
+import { Input } from '../../components/utils/input';
+import { Picker } from '@react-native-picker/picker';
+import { useMutation } from '@apollo/client';
+import { createUser } from '../../components/utils/graphql-requests';
+import { signUpValidations } from './user-signup-validations';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { Navigation } from 'react-native-navigation';
+
+export const UserSignUpScreen = (props: { componentId: string }) => {
+  const [name, setName] = useState('');
+  const [phone, setPhone] = useState('');
+  const [birthDate, setBirthDate] = useState(new Date());
+  const [showBirthDate, setShowBirthDate] = useState('');
+
+  const [email, setEmail] = useState('');
+  const [role, setRole] = useState(UserRoler.user);
+
+  const [message, setMessage] = useState('');
+  const [showMessage, setShowMessage] = useState(false);
+
+  const [show, setShow] = useState(false);
+
+  const [signUp, { loading }] = useMutation(createUser);
+
+  const changeNavigation = () => {
+    Navigation.push(props.componentId, {
+      component: {
+        name: 'UserListScreen',
+      },
+    });
+  };
+
+  const signUpRequest = async () => {
+    try {
+      await signUp({
+        variables: {
+          name: name,
+          phone: phone,
+          birthDate: birthDate.toISOString().split('T')[0],
+          email: email,
+          role: UserRoler[role],
+        },
+      });
+      changeNavigation();
+    } catch (err) {
+      setMessage(err.message);
+    }
+  };
+
+  const handlePress = async () => {
+    const [msg, valid] = signUpValidations(name, phone, birthDate, email);
+    setMessage(msg);
+    if (valid) {
+      signUpRequest();
+    }
+    setShowMessage(true);
+  };
+
+  const onChangeHandler = (event: Event, selectedDate: Date | undefined) => {
```

Comment:
```suggestion
  const handleDatePickerChange = (event: Event, selectedDate: Date | undefined) => {
```

Atenção ao sufixo da convenção. E assim acho que fica mais claro do que se trata esse handle, certo?

Reply:
Feito! Irei me atentar mais as convenções das variáveis